test/src/test/java/org/corfudb/integration/SecurityIT.java

```diff
@@ -0,0 +1,220 @@
+     * each test by converting them from the values provided in CorfuDB.properties. Note that it
+     * will throw {@link NumberFormatException} if the numerical properties provided in CorfuDB.properties
+     * are not well-formed.
+     */
+    @Before
+    public void loadProperties() {
+        // Load host and port properties
+        corfuSingleNodeHost = PROPERTIES.getProperty("corfuSingleNodeHost");
+        corfuStringNodePort = Integer.parseInt(PROPERTIES.getProperty("corfuSingleNodePort"));
+
+        singleNodeEndpoint = String.format("%s:%d",
+                corfuSingleNodeHost,
+                corfuStringNodePort);
+
+        // Load TLS configuration, keystore and truststore properties
+        tlsEnabled = Boolean.valueOf(PROPERTIES.getProperty("tlsEnabledSecurityIT"));
+        serverPathToKeyStore = getPropertyAbsolutePath("serverPathToKeyStore");
+        serverPathToKeyStorePassword = getPropertyAbsolutePath("serverPathToKeyStorePassword");
+        serverPathToTrustStore = getPropertyAbsolutePath("serverPathToTrustStore");
+        serverPathToTrustStorePassword = getPropertyAbsolutePath("serverPathToTrustStorePassword");
+        runtimePathToKeyStore = getPropertyAbsolutePath("runtimePathToKeyStore");
+        runtimePathToKeyStorePass = getPropertyAbsolutePath("runtimePathToKeyStorePass");
+        runtimePathToTrustStore = getPropertyAbsolutePath("runtimePathToTrustStore");
+        runtimePathToTrustStorePass = getPropertyAbsolutePath("runtimePathToTrustStorePass");
+    }
+
+    // Take the property string provided int CorfuDB.properties and return the absolute path string
+    private String getPropertyAbsolutePath(String pathProperty) {
+        return Paths.get(PROPERTIES.getProperty(pathProperty)).toAbsolutePath().toString();
+    }
+
+    /**
+     * This test creates Corfu runtime and a single Corfu server according to the configuration
+     * provided in CorfuDB.properties. Corfu runtime configures TLS related parameters using
+     * {@link CorfuRuntime}'s API and then asserts that operations on a CorfuTable is executed
+     * as Expected.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testServerRuntimeTlsEnabledMethod() throws Exception {
+        // Run a corfu server
+        Process corfuServer = runSinglePersistentServerTls();
+
+        // Start a Corfu runtime
+        CorfuRuntime corfuRuntime = new CorfuRuntime(singleNodeEndpoint)
+                                    .enableTls(runtimePathToKeyStore,
+                                               runtimePathToKeyStorePass,
+                                               runtimePathToTrustStore,
+                                               runtimePathToTrustStorePass)
+                                    .setCacheDisabled(true)
+                                    .connect();
+
+        // Create CorfuTable
+        CorfuTable testTable = corfuRuntime
```

Comment:
We can even move this to AbstractIT. `createTable`
It will simplify a lot of our code. Just a suggestion.

Reply:
I think it provides easier to follow if we don't have many indirection in tests like this. Indirections add one layer of complexity and might not justify their weight specifically in tests. If you don't mind, I'll keep it.